lib/msteams_hermes/message.rb

```diff
@@ -9,6 +9,29 @@ module MsTeamsHermes
   # https://docs.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-using?tabs=cURL#send-adaptive-cards-using-an-incoming-webhook
   ##
   class Message
+    ##
+    # Raises when the message is larger than the latest known maximum size of microsoft teams messages
+    ##
+    class MessageBodyTooLargeError < StandardError
+      def initialize(current_size)
+        super "Microsoft Teams Webhook answered with a 413 due to content size limitations" \
+              " (last check it was about #{MSTEAMS_MESSAGE_SIZE_LIMIT} bytes)." \
+              "\nYour message results in a size of about #{current_size} bytes which is too much."
+      end
+    end
+
+    ##
+    # Raises when the response message changes from the successful response
+    ##
+    class UnknownError < StandardError
+      def initialize(error)
+        super "Microsoft Teams Webhook had an unexpected response body:\n#{error}"
+      end
+    end
+
+    MSTEAMS_MESSAGE_SIZE_LIMIT = 21_000
```

Comment:
Do you have a reference for that limit? It would be great to add that reference.

Reply:
There is a limit stated here: https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/add-incoming-webhook
But it did not seemed to match the actual results of running (I tried multiple times)